shared/util/debug.tsx

```diff
@@ -66,3 +69,40 @@ export function createLoggingProxy<T extends {[key: string]: unknown}>(
 
   return proxy as T
 }
+
+const maybeDebugger = () => {
+  if (debuggerOnWrapError) {
+    // eslint-disable-next-line no-debugger
+    debugger
+  }
+}
+
+export function wrapErrors<T extends (...args: Array<any>) => any>(f: T, logExtra: string = ''): T {
+  return ((...p: Parameters<T>): ReturnType<T> => {
+    try {
+      const result = f(...p)
+      if (result instanceof Promise) {
+        return result.catch(e => {
```

Comment:
tricky. have to handle async function returning promises, which won't get caught below